File: Python/signal.py
#Python3 code for signal file generation
import numpy as np

# Pre-code
Fs = 100
Ts = 1 / Fs
Bits = 8  # Bits used in conversion

# Time Vector
t = np.arange(0, 1, Ts)

# Signal
m = np.sin(2 * np.pi * t) + 2.5 * np.cos(2 * np.pi * t * 4 + np.pi / 7)
import matplotlib.pyplot as plt

plt.plot(t, m)
plt.grid()
plt.title("Signal")
plt.xlabel("Time [s]")
plt.ylabel("Amplitude [V]")
plt.show()

# Conditioning
m = m + abs(np.min(m))  # Positive Offset
Out = (m / np.max(m)) * ((2**Bits) - 1)  # Normalization and Codification
# Exporting
Outbin = [format(int(round(value)), '08b') for value in Out]  # Decimal to Binary conversion

with open(r'/home/tiago/Downloads/signal.txt', 'w') as fp:
        fp.write("\n".join(str(item) for item in Outbin))
 

File: Matlab/signal.m
%Pre-code 
    clc,clear all,close all;
%
%Sampling frequency
    Fs = 100;
    Ts = 1/Fs;
    Bits = 8;%Bits used in conversion
%Time Vector
    t = 0:Ts:1-Ts;
%Signal
    m = sin(2*pi*t)+2.5*cos(2*pi*t*4 + pi/7);
    figure(),plot(t,m),grid,title("Signal"),xlabel("Time [s]"),ylabel("Amplitute [V]");
%
%Conditioning
    m = m + abs(min(m));%Positive Offset 
    Out = (m/max(m))*((2^Bits)-1);%Normalization and Codification
%
%Exporting
    Outbin = dec2bin(Out); %Decimal to Binary conversion
    fid = fopen('signal.txt','wt');%Open or create 'signal.txt' file in write mode
    fprintf(fid,'%s\n',string(Outbin));%Print binary values 8 bits per line
    fclose(fid);%Save file,close file,brings peace to the world
%
  
